File: frontend/ckd-care-app/src/api/dietSurvey.ts
import { api } from "./client";

export interface DietSurveyCreateRequest {
  surveyed_date: string;
  soup_stew_per_day: number;
  sweet_drink_per_day: number;
  fried_food_per_week: number;
  vegetables_every_meal: boolean;
  // 칼륨·단백질: A(G1)·B(G2)·진단자만 응답 (음식군 빈도 0적음/1보통/2많음). 미해당·미응답은 null.
  potassium_food_freq?: number | null;
  protein_food_freq?: number | null;
}

export interface DietSurveyResponse {
  id: number;
  user_id: number;
  surveyed_date: string;
  soup_stew_per_day: number;
  sweet_drink_per_day: number;
  fried_food_per_week: number;
  vegetables_every_meal: boolean;
  created_at: string;
}

export interface DietSurveyListResponse {
  total: number;
  items: DietSurveyResponse[];
}

export interface SurveyStatusResponse {
  lifestyle_survey: boolean;
  diet_survey: boolean;
}

export const dietSurveyApi = {
  create: (body: DietSurveyCreateRequest) =>
    api.post<DietSurveyResponse>("/diet-surveys", body),
  list: (limit = 20, offset = 0) =>
    api.get<DietSurveyListResponse>(`/diet-surveys?limit=${limit}&offset=${offset}`),
  getSurveyStatus: () =>
    api.get<SurveyStatusResponse>("/surveys/status"),
};

File: app/services/diet_survey.py
from app.dtos.diet_survey import (
    DietSurveyCreateRequest,
    DietSurveyListResponse,
    DietSurveyResponse,
    SurveyStatusResponse,
)
from app.repositories.diet_survey_repository import DietSurveyRepository
from app.repositories.lifestyle_survey_repository import LifestyleSurveyRepository


class DietSurveyService:
    def __init__(self) -> None:
        self._repo = DietSurveyRepository()
        self._lifestyle_repo = LifestyleSurveyRepository()

    async def create_survey(self, user_id: int, dto: DietSurveyCreateRequest) -> DietSurveyResponse:
        survey = await self._repo.create(
            user_id=user_id,
            surveyed_date=dto.surveyed_date,
            soup_stew_per_day=dto.soup_stew_per_day,
            sweet_drink_per_day=dto.sweet_drink_per_day,
            fried_food_per_week=dto.fried_food_per_week,
            vegetables_every_meal=dto.vegetables_every_meal,
            potassium_food_freq=dto.potassium_food_freq,
            protein_food_freq=dto.protein_food_freq,
        )
        return DietSurveyResponse.model_validate(survey)

    async def get_surveys(self, user_id: int, limit: int = 20, offset: int = 0) -> DietSurveyListResponse:
        total, items = await self._repo.get_by_user(user_id, limit, offset)
        return DietSurveyListResponse(
            total=total,
            items=[DietSurveyResponse.model_validate(s) for s in items],
        )

    async def get_survey_status(self, user_id: int) -> SurveyStatusResponse:
        lifestyle_done = await self._lifestyle_repo.exists_by_user(user_id)
        diet_done = await self._repo.exists_by_user(user_id)
        return SurveyStatusResponse(lifestyle_survey=lifestyle_done, diet_survey=diet_done)
